File: src/main/java/com/imcode/imcms/mapping/dao/IPAccessDao.scala
package com.imcode.imcms.mapping.dao

import com.imcode._
import com.imcode.imcms.mapping.orm.IPAccess
import org.springframework.transaction.annotation.Transactional
import com.imcode.imcms.dao.hibernate.HibernateSupport

@Transactional(rollbackFor = Array(classOf[Throwable]))
class IPAccessDao extends HibernateSupport {

    def getAll(): JList[IPAccess] = hibernate.listAll()

    def delete(id: JInteger) = hibernate.bulkUpdateByNamedParams(
        "DELETE FROM IPAccess i WHERE i.id = :id", "id" -> id
    )

    def save(ipAccess: IPAccess) = hibernate.saveOrUpdate(ipAccess)

    def get(id: JInteger) = hibernate.get[IPAccess](id)
}

File: src/main/java/com/imcode/imcms/mapping/dao/IPAccessDao.java
package com.imcode.imcms.mapping.dao;

import com.imcode.imcms.mapping.orm.IPAccess;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import java.util.List;

@Repository
public interface IPAccessDao extends JpaRepository<IPAccess, Integer> {
}
